internal/migrations/20201215140033_non_nil_disk_eligibility_test.go

```diff
@@ -0,0 +1,109 @@
+package migrations
+
+import (
+	"encoding/json"
+	"github.com/go-openapi/strfmt"
+	"github.com/google/uuid"
+	"github.com/jinzhu/gorm"
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/openshift/assisted-service/models"
+	gormigrate "gopkg.in/gormigrate.v1"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("nonNilDiskEligibility", func() {
+	var (
+		db        *gorm.DB
+		gm        *gormigrate.Gormigrate
+		clusterID strfmt.UUID
+		err       error
+	)
+
+	BeforeEach(func() {
+		db = common.PrepareTestDB("non_nil_disk_eligibility")
+		gm = gormigrate.New(db, gormigrate.DefaultOptions, all())
+
+		someString := "arbitrary"
+		someUuid := strfmt.UUID("arbitrary")
+
+		// create cluster in order to get rows from DB
+		clusterID = strfmt.UUID(uuid.New().String())
+		cluster := common.Cluster{Cluster: models.Cluster{
+			ID: &clusterID,
+			Hosts: []*models.Host{
+				// Create an inventory with two empty disks (just like before the migration, the eligibility struct
+				// is also empty)
+				{Inventory: `
+{
+	"disks": [
+		{
+		},
+		{
+		}
+	]
+}
+`,
+					Kind:       &someString,
+					ID:         &someUuid,
+					Href:       &someString,
+					Status:     &someString,
+					StatusInfo: &someString},
+			},
+		}}
+		err = db.Create(&cluster).Error
+		Expect(err).NotTo(HaveOccurred())
+
+		err = gm.MigrateTo("20201215140033")
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	AfterEach(func() {
+		common.DeleteTestDB(db, "non_nil_disk_eligibility")
+	})
+
+	ValidateDisks := func(migrationOccurred bool) {
```

Comment:
I think you can define this as a regular function.

Reply:
Also it might help readability if you took the validation function as an argument. Then the caller could look like:
`expectAllDiskEligibilityTo(BeNil())` or `expectAllDiskElegibilityTo(Not(BeNil())`